### PR TITLE
LPS-139522 Refactor JournalArticleIndexer as contributors

### PR DESCRIPTION
**Pull request by lfbesada**
https://issues.liferay.com/browse/LPS-139522

This pull request is here only to be reviewed by Echo Team, it should not be forwarded, since a part of it must be reviewed by the Search Team. The Search Team will be prompted to review the affected party once the echo review passes.

Previous PR: https://github.com/liferay-echo/liferay-portal/pull/7145

### CODE REVIEW

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

**Comment by lfbesada** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/spi/model/index/contributor/JournalArticleModelDocumentContributor.java [hunk 1]
```diff
@@ -72,6 +72,8 @@ public void contribute(Document document, JournalArticle journalArticle) {
 
 		document.addKeywordSortable(Field.ARTICLE_ID, articleId);
 
+		document.addKeyword(Field.UUID, journalArticle.getUuid());
```
Comment (ealonso) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/spi/model/index/contributor/JournalArticleModelDocumentContributor.java:75]: @lfbesada Why it is necessary? I don't see it in any other ModelDocumentContributor

Comment (lfbesada) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/spi/model/index/contributor/JournalArticleModelDocumentContributor.java:75]: @ealonso  I was comparing in the index the documents indexed with the old indexer and with the contributor's indexation and I found that this field was missing. I was looking for another way to add it but I didn't find any, so I added it in the point that seems to be proper.

**Comment by ealonso** [comment]
https://github.com/lfbesada/liferay-portal/pull/54

**Pull request closed without merging**